clang/lib/Driver/ToolChains/Gnu.cpp

```diff
@@ -677,8 +677,17 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   }
 
   Args.AddAllArgs(CmdArgs, options::OPT_T);
+  bool useLLD = false;
+  const char *Exec = Args.MakeArgString(ToolChain.GetLinkerPath(&useLLD));
+  if (useLLD) {
```

Comment:
Why not `if (Args.hasArg(options::OPT_fuse_ld_EQ) && Args.getLastArgValue(options::OPT_fuse_ld_EQ).ends_with_insensitive("lld"))`?

Reply:
Yeah, mainly to support when someone isn't passing `--fuse-ld=lld`. It should still catch it if clang was configured to use `lld` by default. We're already calling `GetLinkerPath`, which is also conveniently supposed to detect when the linker is `lld` either through clang being configured to use lld by default, `--fuse-ld=lld`, or `--ld-path=<filepath>/ld.lld`.